bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/WlanThermoNanoHandler.java

```diff
@@ -0,0 +1,203 @@
+                    "Could not connect to WlanThermo at " + config.getIpAddress());
+            if (pollingScheduler != null) {
+                pollingScheduler.cancel(false);
+            }
+            pollingScheduler = scheduler.schedule(this::checkConnection, config.getPollingInterval(), TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            State s = wlanThermoNanoCommandHandler.getState(channelUID, data, settings);
+            if (s != null)
+                updateState(channelUID, s);
+        } else {
+            if (wlanThermoNanoCommandHandler.setState(channelUID, command, data)) {
+                logger.debug("Data updated, pushing changes");
+                push();
+            } else {
+                logger.warn("Could not handle command of type {} for channel {}!", command.getClass().toGenericString(),
+                        channelUID.getId());
+            }
+        }
+    }
+
+    private void update() {
+        try {
+            // Update objects with data from device
+            String json = httpClient.GET(config.getUri("/data")).getContentAsString();
+            data = gson.fromJson(json, Data.class);
+            logger.debug("Received at /data: {}", json);
+            json = httpClient.GET(config.getUri("/settings")).getContentAsString();
+            settings = gson.fromJson(json, Settings.class);
+            logger.debug("Received at /settings: {}", json);
+
+            // Update channels
+            for (Channel channel : thing.getChannels()) {
+                State state = wlanThermoNanoCommandHandler.getState(channel.getUID(), data, settings);
+                if (state != null) {
+                    updateState(channel.getUID(), state);
+                } else {
+                    // if we could not obtain a state, try trigger instead
+                    String trigger = wlanThermoNanoCommandHandler.getTrigger(channel.getUID(), data);
+                    if (trigger != null) {
+                        triggerChannel(channel.getUID(), trigger);
+                    }
+                }
+            }
+        } catch (URISyntaxException | InterruptedException | ExecutionException | TimeoutException e) {
+            logger.debug("Update failed, checking connection", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Update failed, reconnecting...");
+            if (pollingScheduler != null) {
+                pollingScheduler.cancel(false);
+            }
+            for (Channel channel : thing.getChannels()) {
+                updateState(channel.getUID(), UnDefType.UNDEF);
+            }
+            checkConnection();
+        }
+    }
+
+    private void push() {
+        for (org.openhab.binding.wlanthermo.internal.api.nano.data.Channel c : data.getChannel()) {
```

Comment:
```suggestion
       data.getChannel().forEach(c -> {
```

Also removes the necessity for the fully qualified class name.

Reply:
Like that, done!